content/docs/testing-recipes.md

```diff
@@ -278,126 +220,106 @@ it("changes value when clicked", () => {
   act(() => {
     jest.advanceTimersByTime(100);
   });
-  expect(yields).toBe([]);
+  expect(handleSelect).not.toHaveBeenCalled();
   act(() => {
-    jest.advanceTimersByTime(1000);
+    jest.advanceTimersByTime(10000);
   });
-  expect(yields).toBe([null]);
+  expect(handleSelect).toHaveBeenCalledTimes(1);
+  expect(handleSelect).toHaveBeenCalledWith(null);
 });
 
-it("should cleanup on being removed", () => {
-  const yields = [];
-  render(
-    <Card onSelect={selection => yields.push(selection)} />
-  );
-  act(() => {
-    jest.advanceTimersByTime(100);
-  });
-  expect(yields).toBe([]);
+test("cleans up on being removed", () => {
+  const handleSelect = jest.fn();
+  const { unmount } = render(<Card onSelect={handleSelect} />);
+  unmount();
   act(() => {
-    render(null);
+    jest.advanceTimersByTime(10000);
   });
-  act(() => {
-    jest.advanceTimersByTime(1000);
-  });
-  expect(yields).toBe([]);
+  expect(handleSelect).not.toHaveBeenCalled();
 });
 
-it("should accept selections", () => {
-  const yields = [];
-  render(
-    <Card onSelect={selection => yields.push(selection)} />
-  );
+test("accepts selections", () => {
+  const handleSelect = jest.fn();
+  const { getByTestId } = render(<Card onSelect={handleSelect} />);
 
-  act(() => {
-    container
-      .querySelector("[data-test-id=2]")
-      .dispatchEvent(new MouseEvent("click", { bubbles: true }));
-  });
+  fireEvent.click(getByTestId("card-button-2"));
 
-  expect(yields).toBe([2]);
+  expect(handleSelect).toHaveBeenCalledTimes(1);
+  expect(handleSelect).toHaveBeenCalledWith(2);
 });
 ```
 
 ### Snapshot testing {#snapshot-testing}
 
-Frameworks like Jest also let you save 'snapshots' of data. Combined with react-test-renderer, we can save component structures, and verify they don't break in the future. It's typically better to make more specific assertions than to use snapshots (or at least ensure that the snapshot is small) because these kinds of tests include implementation details meaning they break easily and teams can get desensitized to snapshot breakages.
+Frameworks like Jest also let you save 'snapshots' of data, like DOM nodes. With snapshot testing, we can save DOM structures, and verify they don't change in the future. It's typically better to make more specific assertions than to use snapshots (or at least ensure that the snapshot is small) because these kinds of tests include implementation details meaning they break easily and teams can get desensitized to snapshot breakages.
 
 ```jsx
 // hello.test.js, again
+import React from "react";
+import { render } from "@testing-library/react";
 import Hello from "./hello";
-import { create, act } from "react-test-renderer";
 
-// a helper to convert a React component tree to json
-function toJSON(element) {
-  let root;
-  act(() => {
-    root = create(element);
-  });
-  return root.toJSON();
-}
+test("preserves its structure", () => {
+  expect(render(<Hello />).container).toMatchInlineSnapshot(`
```

Comment:
By using `toMatchInlineSnapshot`, people avoid one of the biggest dangers of snapshots: "Huge snapshots that nobody reviews." If your snapshot is inline, then you're less likely to let it get huge.

Reply:
(working on a lint rule to enforce snapshot size for inline snaps https://github.com/jest-community/eslint-plugin-jest/issues/244)